File: app-python/entitlements_app/requests/start_subscription_request.py
"""Pydantic request model for command `StartSubscription`."""

from pydantic import BaseModel


class StartSubscriptionRequest(BaseModel):
    accountId: str
    currentSeats: int
    planId: str
    renewsAt: str
    seatLimit: int
    startedAt: str
    status: str
    subscriptionId: str
    tenantId: str

    @property
    def account_id(self) -> str:
        return self.accountId

    @property
    def current_seats(self) -> int:
        return self.currentSeats

    @property
    def plan_id(self) -> str:
        return self.planId

    @property
    def renews_at(self) -> str:
        return self.renewsAt

    @property
    def seat_limit(self) -> int:
        return self.seatLimit

    @property
    def started_at(self) -> str:
        return self.startedAt

    @property
    def subscription_id(self) -> str:
        return self.subscriptionId

    @property
    def tenant_id(self) -> str:
        return self.tenantId

File: app/src/main/java/com/example/spring/StartSubscriptionRequest.java
package com.example.spring;

/**
 * Request DTO for command StartSubscription.
 */
public record StartSubscriptionRequest(
    @jakarta.validation.constraints.NotBlank String accountId,
    int currentSeats,
    @jakarta.validation.constraints.NotBlank String planId,
    @jakarta.validation.constraints.NotBlank String renewsAt,
    int seatLimit,
    @jakarta.validation.constraints.NotBlank String startedAt,
    @jakarta.validation.constraints.NotBlank String status,
    @jakarta.validation.constraints.NotBlank String subscriptionId,
    @jakarta.validation.constraints.NotBlank String tenantId
) {}
